pkgs/development/tools/fastddsgen/default.nix

```diff
@@ -35,7 +35,7 @@ let
 
     outputHashAlgo = "sha256";
     outputHashMode = "recursive";
-    outputHash = "sha256-wnnoyaO1QndAYrqmYu1fO6OJrP1NQs8IX4uh37dVntY=";
+    outputHash = "sha256-ZGWTK665wIX/Biz4JDrbaU4EZNqT7Q8o6DSpziUd/SM=";
   };
 in
 stdenv.mkDerivation {
```

Comment:
How about taking the opportunity to "upgrade" the derivation and use the `finalAttrs` pattern ?

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

Reply:
Thanks for letting me know about `finatAttrs`. I've updated the package.

Reply:
Please use a separate commit for doing it.

Reply:
Thanks for letting me know about finatAttrs. I've updated my other package, but here it seems to not bring any advantage. The only place, where it would have sense is overriding `version` attribute, but after doing that, one would have to update two hashes, one of which is in private `deps` derivation. Therefore, there would be no way to update it by overriding.

It might be possible to refactor the code to make overriding easier, but given that there is an ongoing effort in #235058 to refactor build support for gradle-built packages, this refactoring would need to be changed after/if that PR is merged. Therefore, I'd wait with changes after #235058 is either finalized or abandoned.

Reply:
Ok fair enough.